src/main/java/connectify/logic/commands/EditPersonCommand.java

```diff
@@ -31,58 +33,76 @@
+            + "[" + PREFIX_COMPANY + "COMPANY] "
             + "[" + PREFIX_NAME + "NAME] "
             + "[" + PREFIX_PHONE + "PHONE] "
             + "[" + PREFIX_EMAIL + "EMAIL] "
             + "[" + PREFIX_ADDRESS + "ADDRESS] "
             + "[" + PREFIX_TAG + "TAG]...\n"
             + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_COMPANY + "1 "
             + PREFIX_PHONE + "91234567 "
             + PREFIX_EMAIL + "johndoe@example.com";
 
     public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
 
+    public static final String MESSAGE_NO_COMPANY_PROVIDED = "No company provided.";
+
     private final Index index;
+
+    private final Index companyIndex;
     private final EditPersonDescriptor editPersonDescriptor;
 
     /**
      * @param index of the person in the filtered person list to edit
      * @param editPersonDescriptor details to edit the person with
      */
-    public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
+    public EditPersonCommand(Index index, Index companyIndex, EditPersonDescriptor editPersonDescriptor) {
         requireNonNull(index);
         requireNonNull(editPersonDescriptor);
 
         this.index = index;
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.companyIndex = companyIndex;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
-
+        List<Company> lastShownList = model.getFilteredCompanyList();
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
+        if (this.companyIndex == null) {
+            throw new CommandException(MESSAGE_NO_COMPANY_PROVIDED);
+        }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
+        Company affiliatedCompanyToEdit = lastShownList.get(companyIndex.getZeroBased());
+        if (index.getZeroBased() >= affiliatedCompanyToEdit.getPersonList().asList().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = affiliatedCompanyToEdit.getPersonList().asList().get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
         if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
         }
 
+        Company deletedPersonFromCompany = affiliatedCompanyToEdit.deletePersonFromCompany(personToEdit);
```

Comment:
Could you consider renaming this variable as "deletedPersonFromCompany" seems like the variable is a Person object when it is a Company object?